src/RiskModule.sol

```diff
@@ -0,0 +1,177 @@
+        return 150;
+    }
+
+    // TODO: To be deleted after the asset specific values are implemented
+    function getLiquidationThresholdHARDCODED(address assetAddress) public view returns (uint16) {
+        return 110;
+    }
+
+    /**
+     * @notice Calculate the minimum collateral factor given the assets
+     * @param assetAddresses The List of token addresses of the assets
+     * @param assetIds The list of corresponding token Ids that needs to be checked
+     * @dev For each token address, a corresponding id at the same index should be present,
+     * for tokens without Id (ERC20 for instance), the Id should be set to 0
+     * @param assetAmounts The list of corresponding amounts of each Token-Id combination
+     * @return minCollateralFactor is the minimum collateral factor of the given assets
+     * @dev CAN BE DEPRECATED OR DELETED: Added here for possible later use, if in later stages it is not used delete.
+     */
+    function calculateMinCollateralFactor(
+        address[] calldata assetAddresses,
+        uint256[] calldata assetIds,
+        uint256[] calldata assetAmounts
+    ) public returns (uint16 minCollateralFactor) {
+        uint256 assetAddressesLength = assetAddresses.length;
+        require(
+            assetAddressesLength == assetIds.length && assetAddressesLength == assetAmounts.length,
+            "RM_CMCF: LENGTH_MISMATCH"
+        );
+        minCollateralFactor = type(uint16).max;
+        address assetAddress;
+        uint16 collFact;
+        for (uint256 i; i < assetAddressesLength;) {
+            assetAddress = assetAddresses[i];
+            collFact = getCollateralFactorHARDCODED(assetAddress);
+            if (collFact < minCollateralFactor) {
+                minCollateralFactor = collFact;
+            }
+            unchecked {
+                ++i;
+            }
+        }
+        return minCollateralFactor;
+    }
+
+    /**
+     * @notice Calculate the weighted collateral value given the assets
+     * @param assetAddresses The List of token addresses of the assets
+     * @param valuesPerAsset The list of corresponding monetary values of each asset address.
+     * @return collateralValue is the weighted collateral value of the given assets
+     */
+    function calculateWeightedCollateralValue(address[] calldata assetAddresses, uint256[] memory valuesPerAsset)
+        public
+        view
+        returns (uint256 collateralValue)
+    {
+        uint256 assetAddressesLength = assetAddresses.length;
+        require(assetAddressesLength == valuesPerAsset.length, "RM_CCV: LENGTH_MISMATCH");
+        address assetAddress;
+        uint256 collFact;
+        for (uint256 i; i < assetAddressesLength;) {
+            assetAddress = assetAddresses[i];
+            collFact = getCollateralFactorHARDCODED(assetAddress);
+            collateralValue += valuesPerAsset[i].mulDivDown(100, uint256(collFact));
```

Comment:
If we define the collateral threshold as the inverse of the collateral factor, we can do the additions and mulitplication in the for loop, and only do one devision after the loop